tests/framework-tests/core/js/FluidIoCTests.js

```diff
@@ -4692,4 +4692,29 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
         root.destroy();
     });
 
+    /** FLUID-6128 failure when issuing IoCSS advice with more than 2 components from root **/
+
+    fluid.defaults("fluid.tests.FLUID6128root", {
+        gradeNames: "fluid.modelComponent"
+    });
+
+    fluid.defaults("fluid.tests.FLUID6128advisor", {
+        gradeNames: "fluid.component",
+        distributeOptions: {
+            record: "fluid.tests.FLUID6128addon",
+            target: "{/ intervening-name fluid.modelComponent}.options.gradeNames"
+        }
+    });
+
+    jqUnit.test("FLUID-6128: Failure to match IoCSS selector with more than 2 components from root", function () {
+        var advisor = fluid.tests.FLUID6128advisor();
+        var freeModel = fluid.modelComponent();
```

Comment:
Shall `freeModel` be verified of not being notified, in other words, add an assertion to check `fluid.componentHasGrade(freeModel, "fluid.tests.FLUID6128addon")` is falsy.